driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -2002,20 +2002,38 @@ public void refreshSchemaAndSignal(final Connection connection, final DefaultRes
             if (logger.isDebugEnabled())
                 logger.debug("Refreshing schema for {}{}", keyspace == null ? "" : keyspace, table == null ? "" : '.' + table);
 
+            maybeRefreshSchemaAndSignal(connection, future, rs, keyspace, table);
+        }
+
+        public void waitForSchemaAgreementAndSignal(final Connection connection, final DefaultResultSetFuture future, final ResultSet rs) {
+            maybeRefreshSchemaAndSignal(connection, future, rs, null, null);
+        }
+
+        private void maybeRefreshSchemaAndSignal(final Connection connection, final DefaultResultSetFuture future, final ResultSet rs, final String keyspace, final String table) {
+            final boolean refreshSchema = (keyspace != null); // if false, only wait for schema agreement
+
             executor.submit(new Runnable() {
```

Comment:
Yes, I agree that multiple concurrent DDL statements could be problematic. The problem in sharing a callable is that if you run two DDL queries 8 seconds apart and agreement does not happen, the second one should still wait 10 seconds, not the 2 seconds remaining on the first callable.
Maybe we should provide a way to disable automatic schema refreshes at the Statement level. Users would issue all their DDL queries without automatic refresh, then trigger a manual refresh at the end (which probably requires a new API).